milestone-3/frontend/src/App.tsx

```diff
@@ -0,0 +1,48 @@
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+import { useEffect, useState } from "react";
+import Navbar from "./components/navbar/navbar";
+import { Home } from "./components/home/home";
+import "./App.css";
+import About from "./components/about/about";
+import RecipePage from "./components/recipepage/recipepage";
+import recipes from "./recipeData";
+import { Recipe } from "./recipeData";
+
+function App() {
+  const [externalRecipes, setExternalRecipes] = useState<Recipe[]>([]);
+
+  useEffect(() => {
+    fetch('http://localhost:3001/recipe')
+   .then( response => response.json() )
+   .then( data => {console.log(data)
+  setExternalRecipes(data)} )
+  }, []);
+
+  return (
+    <BrowserRouter>
+      <Navbar />
+      <Routes>
+        <Route path="/" element={<Home />} />
+        <Route path="/about" element={<About />} />
+        {externalRecipes.map((rec) => {
+          return (
+            <Route
+              path={`recipe/:${rec.name}`}
```

Comment:
you don't need to include the colon here if you don't want, that's just how we specify a urlParam but you used props so it isn't necessary (of course it is fine to keep it if you want